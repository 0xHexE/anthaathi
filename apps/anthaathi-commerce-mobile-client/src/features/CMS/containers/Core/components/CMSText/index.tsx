import {StyleProp, TextStyle, StyleSheet, View} from 'react-native';
import React from 'react';
import {CoreComponentType} from '../../../../types/common';
import {Text} from 'react-native-paper';

const CMSText = ({
  title,
  textStyle,
  variant = 'titleLarge',
}: {
  title: string;
  textStyle?: StyleProp<TextStyle>;
  variant?:
    | 'displayLarge'
    | 'displayMedium'
    | 'displaySmall'
    | 'headlineLarge'
    | 'headlineMedium'
    | 'headlineSmall'
    | 'titleLarge'
    | 'titleMedium'
    | 'titleSmall'
    | 'labelLarge'
    | 'labelMedium'
    | 'labelSmall'
    | 'bodyLarge'
    | 'bodyMedium'
    | 'bodySmall';
}) => {
  return (
    <View testID="CMSText">
      <Text
        variant={variant}
        style={StyleSheet.flatten([
          {
            fontSize: 24,
            fontWeight: '700',
            textAlign: 'center',
            marginVertical: 10,
          },
          textStyle,
        ])}>
        {title}
      </Text>
    </View>
  );
};

export default CMSText;

export const TextCMSInput = {
  _component: CoreComponentType.CMSText,
  component: CMSText,
};
