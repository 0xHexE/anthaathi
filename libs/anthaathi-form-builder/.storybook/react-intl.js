const locales = ['en-US'];

const messages = locales.reduce(
  (acc, lang) => ({
    ...acc,
    [lang]: import(`../locales/compiled-locales/${lang}.json`), // whatever the relative path to your messages json is
  }),
  {},
);

const formats = {};

export const reactIntl = {
  defaultLocale: 'en-US',
  locales,
  messages,
  formats,
};
