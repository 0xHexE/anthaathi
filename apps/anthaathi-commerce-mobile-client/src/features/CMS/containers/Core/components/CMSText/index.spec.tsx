import React from 'react';
import 'react-native';
import {render} from '@testing-library/react-native';
import {IntlProvider} from 'react-intl';

import locale from '../../../../../../compiled-locales/en-US.json';
import {ThemeProvider} from 'react-native-paper';
import CMSText from '.';

describe('CMSText', () => {
  it('should render the item', function () {
    const temp = render(
      <ThemeProvider>
        <IntlProvider locale="en-US" messages={locale}>
          <CMSText title="title" />
        </IntlProvider>
      </ThemeProvider>,
    );

    expect(temp).toMatchSnapshot();
    expect(temp.queryByTestId('CMSText')).toBeTruthy();
  });

  it('should have CMSText title', () => {
    const temp = render(
      <ThemeProvider>
        <IntlProvider locale="en-US" messages={locale}>
          <CMSText title="title" />
        </IntlProvider>
      </ThemeProvider>,
    );

    expect(temp.queryByText('title')).toBeTruthy();
  });
});
