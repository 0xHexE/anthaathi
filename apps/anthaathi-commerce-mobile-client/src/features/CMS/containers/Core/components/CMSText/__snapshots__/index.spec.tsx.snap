// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CMSText should render the item 1`] = `
<View
  testID="CMSText"
>
  <Text
    style={
      [
        {
          "color": "#000000",
          "fontFamily": "System",
          "fontWeight": "400",
        },
        {
          "textAlign": "left",
        },
        {
          "fontSize": 24,
          "fontWeight": "700",
          "marginVertical": 10,
          "textAlign": "center",
        },
      ]
    }
    variant="titleLarge"
  >
    title
  </Text>
</View>
`;
